sstables/generation_type.hh

```diff
@@ -18,21 +18,26 @@
 namespace sstables {
 
 class generation_type {
-    int64_t _value;
+public:
+    using int_t = int64_t;
```

Comment:
nit: Since the intention is to 

> it can be used everywhere to prepare for uuid sstable generation support.

then maybe it's worth giving it a `value_t` name from the very beginning? Other wise uuid-switching PR would need to `s/int_t/something_else_t/`

Reply:
i think `int_t` only matters where we expect a integer-like identifier, and we want to hide the exact integer type from user. in the uuid-switching PR, i think we will have a simpler way for accessing the underlying integer identifier, either use a `explicit operator int64_t()` or `get<int64_t>()`. IMHO, `int_t` is an intermediate solution before we have uuid-based generation identifier.

Reply:
I tried to get rid of it as much as possible in this series by converting to abstract `sstables::generation_type`
those place that are left deal only with the `int_t`.
With UUID support I imagine that the `value_type` would be a `std::variant<int_t, uuid_t>`

Reply:
value_type is the common name for this type of thing.

Reply:
Maybe it could be gone once when we finish the transition to UUID, but I'd still like places that use a numeric generation to use a typedef, not plain int64_t/unsigned long/long/unsigned type(s) like we use ad-hoc today all over the place.

Reply:
I'm contemplating even to use a strong type based on the tagged_integral class I proposed in https://github.com/scylladb/scylladb/pull/12966/commits/abb832238c015ab7428d95d363f058a923c7e352